Core/Resource/SymbolJson.cs

```diff
@@ -30,218 +30,215 @@ public static void WriteObject(this JsonTextWriter writer, string name, object v
-                            Writer.WriteStartObject();
-                            Writer.WriteObject("Type", output.OutputData.DataType);
-                            output.OutputData.ToJson(Writer);
-                            Writer.WriteEndObject();
+                            writer.WritePropertyName(JsonKeys.OutputData);
+                            writer.WriteStartObject();
+                            writer.WriteObject(JsonKeys.Type, output.OutputData.DataType);
+                            output.OutputData.ToJson(writer);
+                            writer.WriteEndObject();
                         }
 
                         if (output.DirtyFlagTrigger != output.OutputDefinition.DirtyFlagTrigger)
                         {
-                            Writer.WriteObject("DirtyFlagTrigger", output.DirtyFlagTrigger);
+                            writer.WriteObject(JsonKeys.DirtyFlagTrigger, output.DirtyFlagTrigger);
                         }
 
                         if (output.IsDisabled)
                         {
-                            Writer.WriteValue("IsDisabled", output.IsDisabled);
+                            writer.WriteValue(JsonKeys.IsDisabled, output.IsDisabled);
                         }
 
-                        Writer.WriteEndObject();
+                        writer.WriteEndObject();
                     }
                 }
 
-                Writer.WriteEndArray();
+                writer.WriteEndArray();
 
-                Writer.WriteEndObject(); // child
+                writer.WriteEndObject(); // child
             }
 
-            Writer.WriteEndArray();
+            writer.WriteEndArray();
         }
         #endregion
 
         #region reading
-        private static SymbolChild ReadSymbolChild(Model model, JToken symbolChildJson)
+        private static bool TryReadSymbolChild(Model model, JToken symbolChildJson, out SymbolChild symbolChild)
         {
-            var childId = Guid.Parse(symbolChildJson["Id"].Value<string>());
-            var symbolId = Guid.Parse(symbolChildJson["SymbolId"].Value<string>());
+            var childId = Guid.Parse(symbolChildJson[JsonKeys.Id].Value<string>());
+            var symbolId = Guid.Parse(symbolChildJson[JsonKeys.SymbolId].Value<string>());
             if (!SymbolRegistry.Entries.TryGetValue(symbolId, out var symbol))
             {
                 // If the used symbol hasn't been loaded so far ensure it's loaded now
-                symbol = model.ReadSymbolWithId(symbolId);
-            }
-
-            if (symbol == null)
-            {
-                Log.Warning($"Failed to load symbol {symbolId}.");
-                return null;
+                var loaded = model.TryReadSymbolWithId(symbolId, out symbol);
+                if (!loaded)
+                {
+                    Log.Warning($"Error loading symbol child {symbolId}");
+                    symbolChild = null;
+                    return false;
+                }

```

Comment:
A single clear point in the function of total failure. If we get past here, we have a valid `Symbol` object.